src/main/java/io/jenkins/plugins/luxair/ImageTag.java

```diff
@@ -133,6 +139,40 @@ private static String getAuthToken(String[] authService, String image, String us
         return token;
     }
 
+    private static List<String> getImageTagsFromGooglePrivateRegistry(String image, String registry, String authType, String token) {
+        List<String> tags = new ArrayList<>();
+        String getTokenUrl = registry + "/v2/token?service=gcr.io&scope=repository:{image}:pull";
+        String getTagsUrl = registry + "/v2/" + image + "/tags/list";
+
+        Unirest.config().reset();
+        Unirest.config().enableCookieManagement(false).interceptor(errorInterceptor);
+        HttpResponse<JsonNode> response = Unirest.get(getTokenUrl)
+            .header("Authorization", authType + " " + token)
+            .routeParam("image", image)
+            .asJson();
```

Comment:
sorry to chime in here, but may I ask if u have a link to the Google documentation for the interaction with the google registries?

Kinda curious, why u can't use the token u already got from `getAuthToken(...)`?

Reply:
$ curl -D - https://gcr.io/v2/
HTTP/2 401
docker-distribution-api-version: registry/2.0
**www-authenticate: Bearer realm="https://gcr.io/v2/token",service="gcr.io"**
content-type: application/json
date: Sun, 26 Jul 2020 12:46:30 GMT
server: Docker Registry
cache-control: private
x-xss-protection: 0
x-frame-options: SAMEORIGIN
alt-svc: h3-29=":443"; ma=2592000,h3-27=":443"; ma=2592000,h3-T050=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
accept-ranges: none
vary: Accept-Encoding

Yes is seems that it use the same auth mechanism (OAuth 2.0)

Reply:
Sorry, here's a cleaner fix. Seems like route params do not support / in image names, which are always used for gcr. I rewrote the whole thing instead...

Reply:
oh nice to know, and much simpler implementation :smile:

Reply:
Yeah, updated the pr description too :weary::grimacing: